packages/task/src/browser/task-service.ts

```diff
@@ -726,27 +726,17 @@ export class TaskService implements TaskConfigurationClient {
      */
     getTaskByTaskIdentifierAndTaskDefinition(taskDefinition: TaskDefinition | undefined, taskIdentifier: TaskIdentifier, tasks: TaskConfiguration[]): TaskConfiguration {
         const identifierProperties: string[] = [];
-        let relevantTasks = tasks.filter(t =>
-            taskDefinition && t.hasOwnProperty('taskType') &&
```

Comment:
you are removing using `taskDefinition` from the `getTaskByTaskIdentifierAndTaskDefinition()` method.
I believe the parameter `taskDefinition` should be removed and the documentation for the method should be updated